sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -462,12 +462,12 @@ case class NewInstance(
       val d = outerObj.getClass +: paramTypes
       val c = getConstructor(outerObj.getClass +: paramTypes)
       (args: Seq[AnyRef]) => {
-        c.newInstance(outerObj +: args: _*)
+        c(Seq(outerObj +: args: _*))
```

Comment:
Do these need to be called as varargs? it's no longer a Constructor but a function accepting a Seq.

Reply:
Ah, good catch. I meant to change this before.